sdk/cosmosdb/cosmos/src/client/V2ChangeFeed/ChangeFeedEpkRange.ts

```diff
@@ -0,0 +1,327 @@
+   *
+   * If yes, it picks the value of children epkRanges from existing partitionKeyRange cache and enqueues them.
+   */
+  private async handleSplit(
+    shiftLeft: boolean,
+    resolvedRanges: any,
+    oldFeedRange: ChangeFeedRange
+  ): Promise<void> {
+    const contToken = oldFeedRange.continuationToken;
+    let flag = 0;
+    if (shiftLeft) {
+      const oldFeedRangeKey = JSON.stringify([
+        oldFeedRange.minInclusive,
+        oldFeedRange.maxExclusive,
+      ]);
+      const newFeedRange = new ChangeFeedRange(
+        resolvedRanges[0].minInclusive,
+        resolvedRanges[0].maxExclusive,
+        contToken
+      );
+
+      this.continuationTokenMap.updateFeedRange(oldFeedRangeKey, newFeedRange);
+      this.queue.modifyFirstElement(newFeedRange);
+      flag = 1;
+    }
+
+    for (let i = flag; i < resolvedRanges.length; i++) {
+      const newFeedRange = new ChangeFeedRange(
+        resolvedRanges[i].minInclusive,
+        resolvedRanges[i].maxExclusive,
+        contToken
+      );
+      this.queue.enqueue(newFeedRange);
+      this.continuationTokenMap.addFeedRange(newFeedRange);
+    }
+  }
+
+  private async getPartitionRangeId(feedRange: ChangeFeedRange): Promise<string> {
+    const resolvedRanges = await this.partitionKeyRangeCache.getOverlappingRanges(
+      this.url,
+      new QueryRange(feedRange.minInclusive, feedRange.maxExclusive, true, false),
+      false
+    );
+
+    const firstResolvedRange = resolvedRanges[0];
+    if (resolvedRanges.length > 1) {
+      await this.handleSplit(true, resolvedRanges, feedRange);
+    }
+    return firstResolvedRange.id;
+  }
+
+  private async getFeedResponse(
+    feedRange: ChangeFeedRange
+  ): Promise<ChangeFeedResponseV2<Array<T & Resource>>> {
+    const feedOptions: FeedOptions = { initialHeaders: {}, useIncrementalFeed: true };
+
+    if (typeof this.changeFeedOptions.maxItemCount === "number") {
+      feedOptions.maxItemCount = this.changeFeedOptions.maxItemCount;
+    }
+
+    if (this.changeFeedOptions.sessionToken) {
+      feedOptions.sessionToken = this.changeFeedOptions.sessionToken;
+    }
+
+    if (feedRange.continuationToken) {
+      feedOptions.accessCondition = {
+        type: Constants.HttpHeaders.IfNoneMatch,
+        condition: feedRange.continuationToken,
+      };
+    }
+
+    if (this.startTime) {
+      feedOptions.initialHeaders[Constants.HttpHeaders.IfModifiedSince] = this.startTime;
+    }
+
+    const rangeId = await this.getPartitionRangeId(feedRange);
+
+    try {
+      const response: Response<Array<T & Resource>> = await (this.clientContext.queryFeed<T>({
+        path: this.resourceLink,
+        resourceType: ResourceType.item,
+        resourceId: this.resourceId,
+        resultFn: (result) => (result ? result.Documents : []),
+        query: undefined,
+        options: feedOptions,
+        partitionKey: undefined,
+        partitionKeyRangeId: rangeId,
```

Comment:
How does this handle the case of a merge where we need to send the EPK Range header? Passing the range id works for scenarios where the EPK Range is the full partition, and for splits, but when a merge happens, the EPK Range now represents a part of a physical partition, in which case the range is still valid but you cannot send the range id, you need to send the EPK Range min/max headers. Was EPK Range routing added to the JS SDK before?